sklearn/naive_bayes.py

```diff
@@ -117,6 +119,320 @@ def predict_proba(self, X):
         return np.exp(self.predict_log_proba(X))
 
 
+class GeneralNB(_BaseNB, _BaseComposition, ClassifierMixin):
+    """Naive Bayes metaclassifier for multiple naive Bayes models
+
+    The General Naive Bayes classifier is a metaestimator that allows
+    different features in the data to be modeled with different naive Bayes
+    models. This is useful for data containing numerical and categorical
+    features, where numerical features may be modeled as Gaussian distributions
+    and categorical features as categorical distributions.
+
+    Read more in the :ref:`User Guide <general_naive_bayes>`.
+
+    Parameters
+    ----------
+    models : list of tuples
+        List of (name, naive Bayes model, column(s)) tuples specifying the
+        naive Bayes models to apply on the corresponding columns.
+        This is similar to :class:`Pipeline <sklearn.pipeline.Pipeline>`
+        or :class:`ColumnTransformer <sklearn.compose.ColumnTransformer>`.
+
+        name : string
+            This is a user-defined identifier that allows the models and its
+            parameters to be retrieved and set later using :meth:`get_params_`
+            and :meth:`set_params_`.
+        naive Bayes model : Estimator
+            The naive Bayes model represents the distribution assumption on
+            the features. Use our naive Bayes estimators like
+            :class:`GaussianNB <sklearn.naive_bayes.GaussianNB>` and
+            :class:`CategoricalNB <sklearn.naive_bayes.CategoricalNB>`.
+            Custom estimators must support :term:`fit`, :term:`predict`
+            and `_joint_log_likelihood`.
+        column(s) : array-like of {string or int}, slice, or callable
+            Features that correspond to the naive Bayes models. Indexes
+            the data on its second axis. Integers are interpreted as
+            positional columns, while strings reference DataFrame columns
+            by name. A callable is passed the input data `X` and must return
+            one of the above. For example,
+            :func:`compose.make_column_selector
+            <sklearn.compose.make_column_selector>`
+            can select multiple columns by name or dtype.
+
+    Attributes
+    ----------
+    models_ : list of tuples
+        Verified list of (name, naive Bayes model, column(s)),
```

Comment:
mention that they have been fitted please

Reply:
Got it.